api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.openmrs.Condition;
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.db.ConditionDAO;
+import org.slf4j.Logger;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+
+/**
+ * This class implements the {@link ConditionService} interface
+ * It defines the methods for the condition domain object
+ *
+ * @since 2.2
+ */
+@Transactional
+public class ConditionServiceImpl extends BaseOpenmrsService implements ConditionService{
+
+	private static final Logger log = org.slf4j.LoggerFactory.getLogger(ConditionServiceImpl.class);
+
+	private ConditionDAO conditionDAO;
+
+	private ConditionService conditionService;
+	
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 */
+	@Override
+	public Condition save(Condition condition) {
+		if(condition.getCondition() == null){
+			throw new APIException("The condition parameter is required");
+		}
+		if(condition.getClinicalStatus() == null){
+			throw new APIException("The clinical status is required");
+		}
+		Condition existingCondition = getConditionByUuid(condition.getUuid());
+		if(condition.equals(existingCondition)){
+			return existingCondition;
```

Comment:
I think this can be misleading, you should throw an exception, possibly move this logic to a validator and let it fail from there